benchmarks/benchmarks/singlediode.py

```diff
@@ -0,0 +1,41 @@
+"""
+ASV benchmarks for singlediode.py
+"""
+# slower MT implementation in numpy<=1.16
+from numpy.random import RandomState
+# replace with below when ASV migrates to numpy>=1.17
+# and replace 'rng.rang()' with 'rng()'
+# from numpy.random import Generator, MT19937
+from pvlib import singlediode as _singlediode
+
+
+def b88(params):
+    # for a fair comparison, need to also compute isc, voc, i_x and i_xx
+    isc = _singlediode.bishop88_i_from_v(0., *params)
+    voc = _singlediode.bishop88_v_from_i(0., *params)
+    imp, vmp, pmp = _singlediode.bishop88_mpp(*params)
+    ix = _singlediode.bishop88_i_from_v(vmp/2., *params)
+    ixx = _singlediode.bishop88_i_from_v((voc + vmp)/2., *params)
+    return imp, vmp, pmp, isc, voc, ix, ixx
+
+
+class SingleDiode:
+
+    def setup(self):
+        seed = 11471
+        rng = RandomState(seed)
```

Comment:
Do we expect this to produce the same random values as we increase our numpy requirement?  If so, do we expect the benchmark times to change if the inputs change?

Reply:
I expect the random values will be the same (same algorithm, but faster implementation). The timing may be very slightly different - most of the execution time is in computing the 10k IV curves.